### PR TITLE
Feature/xinnanli/naming empty string

### LINKED ISSUE #553: Settings: Household naming format: If a field included in format is empty for a contact, an extra "and" appears to the user in the account/household name
Note- this only occurs if there is more than 1 contact in a household. 

Steps to duplicate:
1. I had my Heda Settings->System, Household account naming format as:
{!Salutation} {!{!Firstname}} {!Lastname} Family Format
*I also enabled/checked- Automatic Household Naming
2. I created one contact
Mr. Bud Jackson
3. The Automatically generated account/household name looks correct. It shows:
"Mr. Bud Jackson Family Format"
4. Under the account/household, clicked create new in Related Contacts Section
5. Created another contact
Mrs. Tammy Jackson
6. The automatically updated household/ account name went to:
"Mrs. and Mr. Tammy and Bud Jackson Family Format"
this looks correct except it can be confusing as Mr. is next to Tammy. 
7. Under the account/household, clicked create new in Related Contacts Section
8. Created another contact (this time leave salutation off and equal to "None")
Barry Jackson
9. It updates account/household name to:
Mrs. and Mr. and Tammy and Bud and Barry Jackson Family Format
**There is an Extra "and" that is showing when salutation is empty. 
Ideally it should update the account/household name to be: 
Mrs. and Mr. Tammy and Bud and Barry Jackson Family Format

see screenshot. 
added extra and for empty salutation string

Another example with non "Other" Formats:
Heda settings for household name with automatic household naming checked/on:
{!{!FirstName}} {!LastName} Family
1. Create a contact- leave firstname blank
I entered for just last name
Perry
2. The household/account name was: 
Perry Family
which is correct.
3. From the account/household page, I Created another contact using the "New" button in the contacts related tab.
I entered for only last name
Perry
4. The household/account name was: 
and Perry Family
which does not look good. 
see screenshot.

no first name

### PR DESCRIPTION
**Pull request by xinnanlisf**
# Critical Changes

# Changes
Fixes household naming bug when all the data in a contact is empty. Use comma instead of "and" to connect names from different contacts
# Issues Closed
Fixes #553
# New Metadata

# Deleted Metadata

# Testing Notes
(In addition to other household naming regression testing)
1. Go to HEDA setting page
2. Enable automatic household naming. Choose household as default account model, "Other" as Household Account Name Format. Fill in the other format with: {!Salutation} {!FirstName} Family.
3. Create a contact: Salutation = "Mr.", FirstName = "TestFN", LastName = "TestLN"
4. Make sure the household account name is Mr. TestFN Family
5. Create another contact under the same household account: Salutation = "", FirstName = "", LastName = "TestLN2"
6. Make sure the household account name is still Mr. TestFN Family
7. Update the second contact to: Salutation = "Ms.", FirstName = "TestFN2", LastName = "TestLN2"
8. Make sure the household account name is now Mr. TestFN and Ms. TestFN2 Family
9. Create another contact under the same household account: Salutation = "Mrs", FirstName = "TestFN3", LastName = "TestLN3"
10. Make sure the household account name is now Mr. TestFN, Ms. TestFN2 and Mrs. TestFN3 Family
11. Make sure other household account naming testing works too

### CODE REVIEW

File: src/classes/UTIL_ACCT_Naming.cls [hunk 1]
```diff
@@ -221,53 +221,86 @@ public class UTIL_ACCT_Naming {
...
+                                }
+                                if (con.get(nameString) != null) {
+                                    conNameString += con.get(nameString);
                                 }
                             }
                         }
+                        conNameString = conNameString.trim();
+                        if (conNameString != null
+                                && conNameString != '') {
+                            nameContactFieldsString += conNameString;
```
Comment (lifewithryan) [inline src/classes/UTIL_ACCT_Naming.cls]: Is this going to cause issues with spacing possibly? I did see that at one point in my testing that I wound up with Mr. and Mrs.Ryan and Carrie Family. So something isn't quite right here yet.

**Comment by dgamboni** [comment]
While the results are inconsistent with the AC's in the PR, the extra "and" has been removed. Therefore, after discussion with team, we'll be closing this Work Item / PR and opening a "Spec/Spike" to determine expected behavior for HH naming (and by extension Administrative Account naming) in terms of syntax, added name order, etc. See testing notes : https://salesforce.quip.com/4QHyA3dabxqK#CUIACAfM5fo

**Comment by xinnanlisf** [comment]
Hi @juliesnow, I noticed this PR's release note had not been reviewed yet. Would you mind take a look please? Thank you! :)

**Comment by xinnanlisf** [comment]
Hi @juliesnow, I added a bullet point to the release note, fyi

**Pull request merged**